frontend/public/components/hpa.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+// eslint-disable-next-line no-unused-vars
+import { K8sResourceKindReference } from '../module/k8s';
+import { ColHead, DetailsPage, List, ListHeader, ListPage } from './factory';
+import { Cog, LabelList, navFactory, ResourceCog, ResourceLink, ResourceSummary, Timestamp } from './utils';
+import { registerTemplate } from '../yaml-templates';
+
+// Pushes to the HPA created by the HPA YAML template.
+registerTemplate('autoscaling/v2beta1.HorizontalPodAutoscaler', `apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: example
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: example
+  minReplicas: 1
+  maxReplicas: 3
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      targetAverageUtilization: 50`);
+
+const HorizontalPodAutoscalersReference: K8sResourceKindReference = 'HorizontalPodAutoscaler';
+
+const { common } = Cog.factory;
+
+const menuActions = [
+  ...common,
+];
+
+export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
+  return <div className="co-m-pane__body">
+    <div className="row">
+      <div className="col-sm-6">
+        <ResourceSummary resource={hpa} showPodSelector={false} showNodeSelector={false} />
+      </div>
+      <div className="col-sm-6">
+        <dl className="co-m-pane__details">
+          <dt>Reference</dt>
```

Comment:
Maybe `Scale Target Reference` or just `Target Reference`?

Reply:
I debated that, but opted for "Reference" because that's what the CLI uses (now whether the CLI is "right" is debatable, but I figured it was better to be consistent).  Yea/nay?  
<img width="641" alt="screen shot 2018-05-31 at 4 34 35 pm" src="https://user-images.githubusercontent.com/895728/40806808-9afa4d74-64f0-11e8-8296-596f4549d2fb.PNG">

Reply:
A more descriptive label gives some context for what a HPA actually does and avoids any confusion around owner references.  I'd guess the CLI is space limited in this case.

I think its slightly better, but its easy enough to change later if we want.

Reply:
+1 to Scale Target

Reply:
"Scale Target" it is.